liby2/src/Y2ProgramComponent.cc

```diff
@@ -419,7 +421,7 @@ void Y2ProgramComponent::sendToExternal(const string& value)
     // result (..) from the input pipe, which is very important. Otherwise the
     // result value would be dropped.
 
-    write(to_external[1], "\n", 1);
+    (void) (write(to_external[1], "\n", 1) == 1);
```

Comment:
wtf? I think you can check result and log if something wrong happen. This way it looks very magic